controllers/component_controller.go

```diff
@@ -470,46 +480,51 @@ func (r *ComponentReconciler) generateGitops(ctx context.Context, req ctrl.Reque
 		return err
 	}
 
-	// Create a temp folder to create the gitops resources in
-	tempDir, err := ioutils.CreateTempPath(component.Name, r.AppFS)
-	if err != nil {
-		log.Error(err, "unable to create temp directory for GitOps resources due to error")
-		return fmt.Errorf("unable to create temp directory for GitOps resources due to error: %v", err)
-	}
-
-	deployAssociatedComponents, err := devfileParser.GetDeployComponents(compDevfileData)
-	if err != nil {
-		log.Error(err, "unable to get deploy components")
-		return err
-	}
+	// Determine if we're using a Kubernetes job for gitops generation, or generating locally
+	localGitopsGen := (r.AllowLocalGitopsGen && component.Annotations["allowLocalGitopsGen"] == "true") || (r.DoLocalGitOpsGen)
```

Comment:
If `AllowLocalGitopsGen` allows certain resources to generate gitops locally then how come we need `r.AllowLocalGitopsGen`? By this if statement the controller needs `r.AllowLocalGitopsGen` to be true always irrespective of the annotation. Am I wrong or something? 🤔

Reply:
Because I didn't want to always allow local GitOps generation, only if we explicitly chose to enable it. 

TBH, this is primarily for testing, since `suite_test.go` sets the fields for Reconcilers, so we can't toggle `DoLocalGitopsGen` between false/true fort the controller tests